module/enrichers.mjs

```diff
@@ -636,6 +637,95 @@ async function enrichSave(config, label, options) {
 
 /* -------------------------------------------- */
 
+/**
+ * Enrich an item use link to roll an item on the selected token.
+ * @param {string[]} config            Configuration data.
+ * @param {string} [label]             Optional label to replace default text.
+ * @returns {Promise<HTMLElement|null>}         An HTML link if the item link could be built, otherwise null.
+ *
+ * @example Use an item from a Name:
+ * ```[[/item Heavy Crossbow]]```
+ * becomes
+ * ```html
+ * <a class="roll-action" data-type="item">
+ *   <i class="fa-solid fa-dice-d20"></i> Heavy Crossbow
+ * </a>
+ * ```
+ *
+ * @example Use an Item from a UUID:
+ * ```[[/item Actor.M4eX4Mu5IHCr3TMf.Item.amUUCouL69OK1GZU]]```
+ * becomes
+ * ```html
+ * <a class="roll-action" data-type="item">
+ *   <i class="fa-solid fa-dice-d20"></i> Bite
+ * </a>
+ * ```
+ *
+ * @example Use an Item from a "Name UUID":
+ * ```[[/item Actor.Akra (Dragonborn Cleric).Item.Mace]]```
+ * becomes
+ * ```html
+ * <a class="roll-action" data-type="item">
+ *   <i class="fa-solid fa-dice-d20"></i> Mace
+ * </a>
+ * ```
+ *
+ * @example Use an Item from a Relative UUID:
+ * ```[[/item .amUUCouL69OK1GZU]]```
+ * becomes
+ * ```html
+ * <a class="roll-action" data-type="item">
+ *   <i class="fa-solid fa-dice-d20"></i> Bite
+ * </a>
+ * ```
+ */
+async function enrichItem(config, label) {
+  const givenItem = config.values.join(" ");
+  // If config is a UUID
+  const itemUuidMatch = givenItem.match(/^(?<synthid>Scene\.\w{16}\.Token\.\w{16}\.)?(?<actorid>Actor\.\w{16})(?<itemid>\.?Item(?<relativeId>\.\w{16}))$/);
+  if (itemUuidMatch) {
+    const ownerActor = itemUuidMatch.groups.actorid.trim();
+    if (!label) {
+      if ((await fromUuid(givenItem)) == null) {
+        console.warn(`Item not found while enriching ${givenItem}.`);
+        return null;
+      } else label = (await fromUuid(givenItem)).name;
+    }
+    return createRollLink(label, { type: "item", rollItemActor: ownerActor, rollItemUuid: givenItem });
+  }
+
+  // If config is a relative Id/Name
+  const relativeIdMatch = givenItem.match(/^\.\w{16}$/);  // .amUUCouL69OK1GZU
+  const copiedIdMatch = givenItem.match(/^\w{16}$/);      // amUUCouL69OK1GZU
+  const relativeNameMatch = givenItem.startsWith(".");    // .Bite
+  if (relativeIdMatch || copiedIdMatch || relativeNameMatch) {
+    const relativeId = relativeIdMatch ? givenItem.substr(1) : givenItem;
+    const foundActor = game.actors.find(actor => actor.items.get(relativeId));
```

Comment:
We shouldn't be doing an actor search like this. Relative IDs should always be relative to the item being enriched. So if I include `[[/item .amUUCouL69OK1GZU]]` it should only check the actor upon which the document is defined, and display an error otherwise.

If we pass the enrichment options into this method we can use `options.relativeTo` to get the document to which the description belongs, and then work up from there until we find an actor. If we don't find one, or an item with that ID doesn't exist on the actor, we return `null` with an error message, otherwise we use that item for the rest of the process.

Reply:
ok cool, didn't know about `options.relativeTo` I'll try to search up on it, but happy to take any tips on implementing it ;)

Reply:
Should be a simple matter of passing `options` into the method like the other enrichers, and then examining `relativeTo` to see if it is an actor (search for that actor for items), an item (search the item's parent if it exists for items), or something else (return `null`, cause there is no way to know where to search).

Reply:
The other enrichers that use `relativeTo` are the Embeds using `config.docs`, but this enricher doesn't have that, so I'm not sure what to pass into `options = { ...options, relativeTo: ??};` (or if that is even what I should be doing)

Reply:
This doesn't have anything to do with `config`. Relative to is part of the `EnrichmentOptions` that core passes in when the enrichment is happening, so all you have to do is in `enrichString` pass in the third `options` parameter and you'll have access to it.

Reply:
Oh. yea I overcomplicated it, or was doing something wrong when I tried that earlier, but yep I'm seeing it now, should be able to make some adjustments today, thanks!